worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/ThreadSafeCache.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * WorldEdit, a Minecraft world manipulation toolkit
+ * Copyright (C) sk89q <http://www.sk89q.com>
+ * Copyright (C) WorldEdit team and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.sk89q.worldedit.sponge;
+
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.entity.living.player.Player;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CopyOnWriteArraySet;
+
+/**
+ * Caches data that cannot be accessed from another thread safely.
+ */
+public class ThreadSafeCache implements Runnable {
+
+    private static final ThreadSafeCache INSTANCE = new ThreadSafeCache();
+    private Set<UUID> onlineIds = Collections.emptySet();
+
+    /**
+     * Get an concurrent-safe set of UUIDs of online players.
+     *
+     * @return a set of UUIDs
+     */
+    public Set<UUID> getOnlineIds() {
+        return onlineIds;
+    }
+
+    @Override
+    public void run() {
+        Set<UUID> onlineIds = new HashSet<>();
+
+        for (Player player : Sponge.getServer().getOnlinePlayers()) {
+            onlineIds.add(player.getUniqueId());
+        }
+
+        this.onlineIds = new CopyOnWriteArraySet<>(onlineIds);
```

Comment:
It can be modified, but isn't always. This saves the set from being copied at each point while constructing.